src/Volume.jsx

```diff
@@ -2,27 +2,56 @@ import React from 'react';
 import { Button } from "@patternfly/react-core/dist/esm/components/Button";
 import { Checkbox } from "@patternfly/react-core/dist/esm/components/Checkbox";
 import { FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
+import { FormHelper } from "cockpit-components-form-helper.jsx";
 import { FormSelect, FormSelectOption } from "@patternfly/react-core/dist/esm/components/FormSelect";
 import { Grid } from "@patternfly/react-core/dist/esm/layouts/Grid";
 import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput";
 import { MinusIcon } from '@patternfly/react-icons';
 import { FileAutoComplete } from 'cockpit-components-file-autocomplete.jsx';
 import cockpit from 'cockpit';
 
+import * as utils from './util.js';
+
 const _ = cockpit.gettext;
 
-export const Volume = ({ id, item, onChange, idx, removeitem, additem, options, itemCount }) =>
+export function validateVolume(volumes, key, validationFailed) {
+    const delta = { ...validationFailed };
+
+    switch (key) {
+    case "hostPath":
+        break;
+    case "containerPath":
+        if (utils.isEmpty(volumes.containerPath))
+            delta.containerPath = _("Required");
+
+        break;
+    }
+
+    return delta;
+}
+
+export const Volume = ({ id, item, onChange, idx, removeitem, additem, options, itemCount, validationFailed, onValidationChange }) =>
     (
         <Grid hasGutter id={id}>
-            <FormGroup className="pf-m-4-col-on-md" label={_("Host path")} fieldId={id + "-host-path"}>
+            <FormGroup className="pf-m-4-col-on-md"
+                label={_("Host path")}
+                fieldId={id + "-host-path"}
+                onBlur={() => utils.dynamicValidate(item, validationFailed, "hostPath", onValidationChange, validateVolume)}
+                onFocus={() => utils.validationClear(validationFailed, "hostPath", onValidationChange)}
+            >
                 <FileAutoComplete id={id + "-host-path"}
                     value={item.hostPath || ''}
                     onChange={ value => onChange(idx, 'hostPath', value) } />
+                <FormHelper helperTextInvalid={validationFailed?.hostPath} />
             </FormGroup>
             <FormGroup className="pf-m-3-col-on-md" label={_("Container path")} fieldId={id + "-container-path"}>
                 <TextInput id={id + "-container-path"}
                     value={item.containerPath || ''}
+                    validated={validationFailed?.containerPath ? "error" : "default"}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20230725-110243-108e1b8a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)